### PR TITLE
Remove jQuery dependency

### PR DESCRIPTION
**Pull request by tmm1**
This removes jQuery as a dependency, in favor of native DOM manipulation methods available on all modern browsers.

### CODE REVIEW

File: src/videojs.markers.js [hunk 1]
```diff
@@ -231,21 +227,26 @@ type Marker = {
...
-          breakOverlay && breakOverlay.find('.vjs-break-overlay-text').html(setting.breakOverlay.text(marker));
+          if (breakOverlay) breakOverlay.querySelector('.vjs-break-overlay-text').innerHTML = setting.breakOverlay.text(marker);
         }
 
-        breakOverlay && breakOverlay.css('visibility', "visible");
+        if (breakOverlay) breakOverlay.style.visibility = "visible";
       } else {
         overlayIndex = NULL_INDEX;
-        breakOverlay && breakOverlay.css("visibility", "hidden");
+        if (breakOverlay) breakOverlay.style.visibility = "hidden";
```
Comment (spchuang) [inline src/videojs.markers.js]: can we keep the original style? `A && ...`
I'm not a big fan of if statement without {}. So you can also do
```
if (A) {
...
}
```

File: src/videojs.markers.js [hunk 2]
```diff
@@ -189,30 +185,30 @@ type Marker = {
...
+          markerTip.style.marginLeft = -parseFloat(markerTip.getBoundingClientRect().width) / 2 - 5 + 'px'
+          markerTip.style.visibility = 'visible'
         }
-
       });
 
-      markerDiv.on('mouseout',() => {
-        !!markerTip && markerTip.css("visibility", "hidden");
+      markerDiv.addEventListener('mouseout',() => {
+        if (!!markerTip) markerTip.style.visibility = "hidden";
```
Comment (spchuang) [inline src/videojs.markers.js]: nit : `if (A) {...}`

File: src/videojs.markers.js [hunk 3]
```diff
@@ -105,31 +104,28 @@ type Marker = {
...
-      }
+      var markerDiv = videojs.createEl('div', {
+        className: `vjs-marker ${marker.class||""}`,
+      }, {
+        'data-marker-key': marker.key,
+        'data-marker-time': setting.markerTip.time(marker)
+      })
+      Object.keys(setting.markerStyle).forEach(key => {
+        markerDiv.style[key] = setting.markerStyle[key]
+      })
```
Comment (spchuang) [inline src/videojs.markers.js]: nit: trailing ;

File: src/videojs.markers.js [hunk 4]
```diff
@@ -105,31 +104,28 @@ type Marker = {
...
-      // add user-defined class to marker
-      if (marker.class) {
-        markerDiv.addClass(marker.class);
-      }
+      var markerDiv = videojs.createEl('div', {
+        className: `vjs-marker ${marker.class||""}`,
+      }, {
+        'data-marker-key': marker.key,
+        'data-marker-time': setting.markerTip.time(marker)
+      })
```
Comment (spchuang) [inline src/videojs.markers.js]: nit: trailing ;

File: src/videojs.markers.js [hunk 5]
```diff
@@ -105,31 +104,28 @@ type Marker = {
...
-        })
-        .attr("data-marker-key", marker.key)
-        .attr("data-marker-time", setting.markerTip.time(marker));
-
-      // add user-defined class to marker
-      if (marker.class) {
-        markerDiv.addClass(marker.class);
-      }
+      var markerDiv = videojs.createEl('div', {
+        className: `vjs-marker ${marker.class||""}`,
```
Comment (spchuang) [inline src/videojs.markers.js]: nit: space between ||.

File: src/videojs.markers.js [hunk 6]
```diff
@@ -189,30 +185,30 @@ type Marker = {
...
       });
     }
 
     function initializeMarkerTip(): void {
-      markerTip = $("<div class='vjs-tip'><div class='vjs-tip-arrow'></div><div class='vjs-tip-inner'></div></div>");
-      videoWrapper.find('.vjs-progress-holder').append(markerTip);
+      markerTip = videojs.createEl('div', {
+        className: 'vjs-tip',
+        innerHTML: "<div class='vjs-tip-arrow'></div><div class='vjs-tip-inner'></div>",
+      })
```
Comment (spchuang) [inline src/videojs.markers.js]: nit: trailing ;

**Comment by spchuang** [comment]
@tmm1 just took a look. I only have some nits but everything else looks good. Can you try testing on both Chrome & FireFox to make sure it doesn't break any existing behaviors? Thanks a lot.

File: src/videojs.markers.js (hunk 1 above)
Comment (tmm1) [inline src/videojs.markers.js]: The && style does not work when right hand side has an assignment.

I can switch to multi line if statement with braces.

**Comment by tmm1** [comment]
@spchuang I addressed your comments, and tested in Safari, Chrome, and Firefox.

**Comment by spchuang** [comment]
awesome!

**Comment by marcelorubini** [comment]
Could you please release a new version with this integration in order to make this no-jquery version available via cdnjs, unkpg, etc? Would appreciate it a lot!

**Comment by spchuang** [comment]
Will take a look this week!

**Comment by spchuang** [comment]
it is pushed to version 0.9

**Comment by mayadi1** [comment]
Perfect

**Pull request merged**